ts/src/base/ws/Cache.ts

```diff
@@ -205,9 +207,65 @@ class ArrayCacheBySymbolById extends ArrayCache {
         this.allNewUpdates = (this.allNewUpdates || 0) + (afterLength - beforeLength)
     }
 }
+class PositionsCache {
+
+    constructor (positions = []) {
+        this.hashmap = {}
+        for (let i = 0; i < positions.length; i++) {
+            this.append (positions[i]);
+        }
+    }
+
+    append (item) {
+        const bySide = this.hashmap[item.symbol] = this.hashmap[item.symbol] || {}
+        if (item.side in bySide) {
+            const reference = bySide[item.side]
+            if (reference !== item) {
+                for (const prop in item) {
+                    reference[prop] = item[prop]
+                }
+            }
+            item = reference
+        } else {
+            bySide[item.side] = item
+        }
+        Object.defineProperty (item, 'seenBy', {
```

Comment:
@pcriadoperez I'm still trying to figure out if the `newUpdates` concept makes sense when talking about `positions. A trade, order, etc is an event that happens at a specific point in time so it makes sense the "old" vs "new" trades, however, positions are more like "continuous events"

Reply:
Conceptually I agree with you. Also that would simplify the code as the new cache type would not be needed.
The reason I iniatially put it in is that binance USDM api only returns positions updates in the case of a user action, so it did function a bit like "new Orders" where `newUpdates` would make sense.
However I'm checking and in most exchanges the positions are constantly streamed and in binance if we switch to implement the portfolio Margin API it also works in the same way. Where as you say newUpdates stops making sense.

I think you are right that it makes sense to remove it. The only drawback I can think of is if we run into any exchanges that function like binance USDM we will only return the `newUpdates` and not provide a full cache.

Let me know if you'd like me to proceed this way :)

Reply:
@pcriadoperez I'm fine with keeping it for now and reevaluate it later if needed